receiver/splunkhecreceiver/receiver.go

```diff
@@ -264,42 +326,49 @@ func (r *splunkReceiver) handleReq(resp http.ResponseWriter, req *http.Request)
 	}
 }
 
-func (r *splunkReceiver) createResourceCustomizer(req *http.Request) func(pdata.Resource) {
+func (r *splunkReceiver) consumeMetrics(ctx context.Context, events []*splunk.Event, resp http.ResponseWriter, req *http.Request) {
+	var resourceCustomer func(resource pdata.Resource)
 	if r.config.AccessTokenPassthrough {
-		if accessToken := req.Header.Get(splunk.HECTokenHeader); accessToken != "" {
-			return func(resource pdata.Resource) {
+		accessToken := req.Header.Get(splunk.HECTokenHeader)
+		if accessToken != "" {
+			resourceCustomer = func(resource pdata.Resource) {
 				resource.Attributes().InsertString(splunk.HecTokenLabel, accessToken)
 			}
 		}
 	}
-	return func(resource pdata.Resource) {}
-}
-
-func (r *splunkReceiver) consumeMetrics(ctx context.Context, events []*splunk.Event, resp http.ResponseWriter, req *http.Request) {
-	md, _ := splunkHecToMetricsData(r.settings.Logger, events, r.createResourceCustomizer(req), r.config)
+	md, _ := splunkHecToMetricsData(r.settings.Logger, events, resourceCustomer, r.config)
 
 	decodeErr := r.metricsConsumer.ConsumeMetrics(ctx, md)
 	r.obsrecv.EndMetricsOp(ctx, typeStr, len(events), decodeErr)
 
 	if decodeErr != nil {
-		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, decodeErr)
+		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, len(events), decodeErr)
 	} else {
 		resp.WriteHeader(http.StatusAccepted)
 		resp.Write(okRespBody)
 	}
 }
 
 func (r *splunkReceiver) consumeLogs(ctx context.Context, events []*splunk.Event, resp http.ResponseWriter, req *http.Request) {
-	ld, err := splunkHecToLogData(r.settings.Logger, events, r.createResourceCustomizer(req), r.config)
+	var resourceCustomer func(resource pdata.Resource)
+	if r.config.AccessTokenPassthrough {
+		accessToken := req.Header.Get(splunk.HECTokenHeader)
+		if accessToken != "" {
+			resourceCustomer = func(resource pdata.Resource) {
+				resource.Attributes().InsertString(splunk.HecTokenLabel, accessToken)
+			}
+		}
+	}
```

Comment:
Just curios why don't reuse existing `createResourceCustomizer` instead of copy/pasting?

Reply:
I was trying to save instantiating a func(), but it's not worth it. Fixed now.